recipe/0001-Complete-RTTI-fix-for-macOS.patch

```diff
@@ -0,0 +1,73 @@
+diff --git i/cmake-modules/AzureVcpkg.cmake w/cmake-modules/AzureVcpkg.cmake
+index e1db45d18..39351c5ba 100644
+--- i/cmake-modules/AzureVcpkg.cmake
++++ w/cmake-modules/AzureVcpkg.cmake
+@@ -111,7 +111,7 @@ macro(az_vcpkg_export targetName macroNamePart dllImportExportHeaderPath)
+     )
+ 
+   # If building a Windows DLL, patch the dll_import_export.hpp
+-  if(WIN32 AND BUILD_SHARED_LIBS)
++  if((WIN32 OR APPLE) AND BUILD_SHARED_LIBS)
+     add_compile_definitions(AZ_${macroNamePart}_BEING_BUILT)
+     target_compile_definitions(${targetName} PUBLIC AZ_${macroNamePart}_DLL)
+ 
+diff --git i/sdk/core/azure-core/CMakeLists.txt w/sdk/core/azure-core/CMakeLists.txt
+index ad83b71d0..11b62fb72 100644
+--- i/sdk/core/azure-core/CMakeLists.txt
++++ w/sdk/core/azure-core/CMakeLists.txt
+@@ -200,6 +200,21 @@ az_vcpkg_export(
+     "azure/core/dll_import_export.hpp"
+   )
+ 
++# Set symbol visibility for macOS to ensure exception symbols are properly exported
++if(APPLE)
++  set(CMAKE_CXX_VISIBILITY_PRESET default)
++  set(CMAKE_VISIBILITY_INLINES_HIDDEN OFF)
++  # Ensure exception symbols are exported for RTTI to work across shared library boundaries
++  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=default")
```

Comment:
Isn't this redundant with `set(CMAKE_CXX_VISIBILITY_PRESET default)` just above?